src/walk.rs

```diff
@@ -221,16 +219,16 @@ fn spawn_receiver(
             let mut mode = ReceiverMode::Buffering;
 
             // Maximum time to wait before we start streaming to the console.
-            let max_buffer_time = config
-                .max_buffer_time
-                .unwrap_or_else(|| time::Duration::from_millis(100));
+            let max_buffer_time = config.max_buffer_time.unwrap_or(DEFAULT_MAX_BUFFER_TIME);
 
             let stdout = io::stdout();
             let mut stdout = stdout.lock();
 
-            let mut num_results = 0;
+            let rx_iter = rx
+                .into_iter()
+                .take(config.max_results.unwrap_or(std::usize::MAX));
```

Comment:
Concerning readability, I think I would prefer a `if` clause here:
``` rs
if let Some(max_results) = config.max_results {
  rx_iter = rx_iter.take(max_results);
}
```
which would get rid of the `usize::MAX` (even if it's not a real problem).

Reply:
Just to make sure: are we 100% sure that the new `.take(…)` implementation does the same thing as the `break` before? It's certainly more elegant, but this functionality is quite crucial. With `--max-results=<NUM>` (or `-1`), we definitely want `fd` to exit early, *immediately* after finding the `N`*th* (or first) result.

Reply:
The conditional `rx_iter = rx_iter.take(...)` doesn't work because the types are different. Only way around that is to have `rx_iter: &mut dyn Iterator`or similar I think.

The `.take` definetely doesn't wait for the rest of the results and just breaks after the specified number of iterations. reading again, I noticed another small deviation though: Originally this only counts the number of successful results, not the `WorkerResult::Error` cases. While this is probably possible to do with an iterator adapter, it is probably easier to just leave it as it was